core/src/autogluon/core/searcher/local_searcher.py

```diff
@@ -121,15 +111,102 @@ def get_best_reward(self):
         else:
             return dict()
+
+    def _get_params_static(self) -> dict:
+        """
+        Gets a dictionary of static key values, where no search space is used and therefore the values are always the same in all configs.
+        """
+        params_static = dict()
+        for key, val in self.search_space.items():
+            if not isinstance(val, Space):
+                params_static[key] = val
+        return params_static
+
+    def _get_params_default(self, params_static: dict) -> dict:
+        """
+        Gets the default config by calling `val.default` on every search space parameter, plus the static key values.
+        """
+        params_default = dict()
+        for key, val in self.search_space.items():
+            if isinstance(val, Space):
+                params_default[key] = val.default
+        params_default.update(params_static)
+        return params_default
+
+    def _get_params_cat_dict(self) -> dict:
+        """
+        Gets the dictionary of pickled category value -> index mapping for Category search spaces.
+        This is used in `self._pickle_config` to map values to idx when pickling the config. This compresses the size of the pkl file.
+        When being later unpickled via `self._unpickle_config`, the idx can be used to get the key value via `self.search_space[key][idx]`.
+        """
+        params_cat_dict = dict()
+        for key, val in self.search_space.items():
+            if isinstance(val, Categorical):
+                cat_map = dict()
+                for i, cat in enumerate(val.data):
+                    cat_pkl = pickle.dumps(cat)
+                    cat_map[cat_pkl] = i
+
+                params_cat_dict[key] = cat_map
+        return params_cat_dict
+
+    def _add_result(self, config: dict, result: float):
+        assert isinstance(result, (float, int)), f"result must be a float or int! Was instead {type(result)} | Value: {result}"
+        config_pkl = self._pickle_config(config=config)
+        self._results[config_pkl] = result
+
+    def _pickle_config(self, config: dict) -> bytes:
+        assert isinstance(config, dict), f"config must be a dict! Was instead {type(config)} | Value: {config}"
+        assert len(config) == len(self._params_order), f'Config length does not match expected params count!\n' \
+                                                       f'Expected: {self._params_order}\n' \
+                                                       f'Actual:   {list(config.keys())}'
+
+        # Note: This code is commented out because it can be computationally and memory expensive if user sends large objects in search space, such as datasets.
+        """
+        for key in self._params_static:
+            assert pickle.dumps(config[key]) == pickle.dumps(self._params_static[key]), \
+                f'Invalid config value for search space parameter "{key}" | Invalid Value: {config[key]} | Expected Value: {self._params_static[key]}'
+        """
```

Comment:
Should we even allow to do this? This would be a misuse of functionality. HPO spaces must be light. Large objects might create bottlenecks for distributed settings. I would suggest raising an error.

Reply:
This is something I might plan to address after NLP refactor is merged. One possibility is to disallow static parameters in search spaces. Currently we don't duplicate these parameters in memory after this PR (previously we were pickling and copying these parameters every trial unnecessarily).